src/core/Akka.NodeTestRunner/Program.cs

```diff
@@ -66,15 +91,80 @@ static void Main(string[] args)
                             specFail.FailureExceptionTypes.Add(ex.GetType().ToString());
                             specFail.FailureMessages.Add(ex.Message);
                             specFail.FailureStackTraces.Add(ex.StackTrace);
+                            _logger.Tell(specFail.ToString());
                             Console.WriteLine(specFail);
+
+                            //make sure message is send over the wire
+                            FlushLogMessages();
                             Environment.Exit(1); //signal failure
+                            return 1;
                         }
-                        sink.Finished.WaitOne();
-                        Environment.Exit(sink.Passed ? 0 : 1);
+
+                        var timedOut = false;
+                        if (!sink.Finished.WaitOne(MaxProcessWaitTimeout)) //timed out
+                        {
+                            var line = string.Format("Timed out while waiting for test to complete after {0} ms",
+                                MaxProcessWaitTimeout);
+                            _logger.Tell(line);
+                            Console.WriteLine(line);
+                            timedOut = true;
+                        }
+
+                        FlushLogMessages();
+                        system.Shutdown();
+                        system.AwaitTermination();
+
+                        Environment.Exit(sink.Passed && !timedOut ? 0 : 1);
+                        return sink.Passed ? 0 : 1;
                     }
                 }
             }
         }
+
+        private static void FlushLogMessages()
+        {
+            try
+            {
+                _logger.GracefulStop(TimeSpan.FromSeconds(2)).Wait();
```

Comment:
We use `GracefulStop` on the actor that writes out to Akka.IO as our means of flushing the transport - any new log messages added after the `FlushLogMessages` method is called will not be pushed onto the transport.